mlir-tensorrt/executor/include/mlir-executor/Support/Allocators.h

```diff
@@ -128,6 +128,94 @@ class PinnedMemoryAllocator {
   std::unique_ptr<BlockEventQueue> pendingBlockEvents;
 };
 
+//===----------------------------------------------------------------------===//
+// OutputAllocator and CustomTensorRTOuputAllocator
+//===----------------------------------------------------------------------===//
+
+//!
+//! Class to allocate memory for outputs with data-dependent shapes. The sizes
+//! of those are unknown so pre-allocation is not possible.
+//!
+class OutputAllocator {
+public:
+  virtual ~OutputAllocator() = default;
+  virtual void setTensorName(const char *tensorName) = 0;
+  virtual void setCurrentMemory(void *currentMemory) = 0;
+  virtual void setOutputSize(const int64_t outputSize) = 0;
+  virtual void *reallocateOutputAsync(char const *tensorName,
+                                      void *currentMemory, uint64_t size,
+                                      uint64_t alignment,
+                                      CudaStream /*stream*/) = 0;
+  virtual void notifyShape(char const *tensorName, const int64_t *dims,
+                           int64_t nbDims) = 0;
+};
+
+
```

Comment:
It's not clear why we need these two classes instead of one class. Why do we need all these different methods? Similarly, what is the tracker class for?

Reply:
I copied this code from one of my Draft MR's. Class `OutputAllocator` here defines the interface that can be overridden by the user (aka. Tripy) to provide the implementation. 
At the same time, we would like to provide a default implementation in the compiler i.e. `CustomTensorRTOutputAllocator`.

Reply:
`OutputAllocatorTracker` binds the lifetime of `OutputAllocator` and allocates the memory it owns to the session.
May be it should remain alive as long as `RuntimeClient` is alive. I am open to suggestions here.